pkg/agent/proxy/types/groupcounter.go

```diff
@@ -84,25 +83,36 @@ func (c *groupCounter) deleteServicePortNameMap(svcNamespacedName string, svcKey
 	}
 }
 
-func (c *groupCounter) Get(svcPortName k8sproxy.ServicePortName, isEndpointsLocal bool) (binding.GroupIDType, bool) {
+func (c *groupCounter) AllocateNotExist(svcPortName k8sproxy.ServicePortName, isEndpointsLocal bool) binding.GroupIDType {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 	key := keyString(svcPortName, isEndpointsLocal)
 	if id, ok := c.groupMap[key]; ok {
-		return id, false
+		return id
 	} else if len(c.recycled) != 0 {
 		id = c.recycled[len(c.recycled)-1]
 		c.recycled = c.recycled[:len(c.recycled)-1]
 		c.groupMap[key] = id
 		c.updateServicePortNameMap(svcPortName.NamespacedName.String(), key)
 		c.groupIDUpdates <- svcPortName.NamespacedName.String()
-		return id, true
+		return id
 	} else {
 		c.groupIDCounter += 1
 		c.groupMap[key] = c.groupIDCounter
 		c.updateServicePortNameMap(svcPortName.NamespacedName.String(), key)
 		c.groupIDUpdates <- svcPortName.NamespacedName.String()
-		return c.groupIDCounter, true
+		return c.groupIDCounter
+	}
+}
+
+func (c *groupCounter) Get(svcPortName k8sproxy.ServicePortName, isEndpointsLocal bool) (binding.GroupIDType, bool) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	key := keyString(svcPortName, isEndpointsLocal)
+	if id, ok := c.groupMap[key]; ok {
```

Comment:
```suggestion
	return c.groupMap[key]
```

Reply:
Updated as the following. `return c.groupMap[key]` cannot pass syntax check.

```
id, exist := c.groupMap[key]
	return id, exist
```